pro/src/screens/IndividualOffer/SummaryScreen/StockSection/RecurrenceSection/__specs__/RecurrenceSection.spec.tsx

```diff
@@ -35,13 +35,27 @@ describe('StockEventSection', () => {
     }
 
     render(<RecurrenceSection stocksStats={stocksStats} departementCode="" />)
+  })
+
+  it("displays 0 when remainingQuantity is 0 but not 'Illimitée'", () => {
+    const stocksStats = {
+      stockCount: 0,
+      remainingQuantity: 0,
+      oldestStock: '2021-01-01T00:00:00+01:00',
+      newestStock: '2021-01-03T00:00:00+01:00',
+    }
+
+    render(<RecurrenceSection stocksStats={stocksStats} departementCode="" />)
 
     expect(screen.queryByText(/Nombre de dates/)).toBeInTheDocument()
-    expect(screen.queryByText('1')).toBeInTheDocument()
+    expect(screen.queryByText('0')).toBeInTheDocument()
     expect(screen.queryByText(/Période concernée/)).toBeInTheDocument()
-    expect(screen.queryByText('le 01/01/2021')).toBeInTheDocument()
+    expect(
+      screen.queryByText('du 01/01/2021 au 03/01/2021')
+    ).toBeInTheDocument()
     expect(screen.queryByText(/Capacité totale/)).toBeInTheDocument()
-    expect(screen.queryByText('637 places')).toBeInTheDocument()
+    expect(screen.queryByText('Illimitée')).not.toBeInTheDocument()
+    expect(screen.queryByText('0 place')).toBeInTheDocument()
```

Comment:
Le test est très bien, c'est exactement ça !

des petits détails, tu peux ne garder que les deux lignes qui concernent ce qu'on voulait tester (le reste est déjà tester dasn les autres tests:
```
    expect(screen.queryByText('Illimitée')).not.toBeInTheDocument()
    expect(screen.queryByText('0 place')).toBeInTheDocument()
```

et une petite subtilité de la libraire de tests qu'on utilise :
- les selecteurs qui commencent par queryQuelqueChose (quand on espère ne pas trouver de résultat, qu'on a mis .not.toBeInTheDocument() derrière
- si on espère avoir un résultat, on utilise  getByQuelqueChose (getByText dans ton cas)

après je me rend compte que c'est moi qui ai fait l'erreur sur les autres tests du fichier
-> corrige au minimum ton test, si tu as du courage tu peux corriger tout le fichier de test :)

Reply:
D'accord Romain, super.